modyn/config/schema/pipeline/evaluation/handler.py

```diff
@@ -33,6 +33,16 @@ class EvalHandlerConfig(ModynBaseModel):
     execution_time: EvalHandlerExecutionTime = Field(
         description="When evaluations should be performed in terms of where in the code / when during a pipeline run."
     )
+    """Caution: execution_time=`after_training` yields slightly different behavior for the `models` option.
+
+    Point of difference: decision, if a model is the most recent one for a certain eval request.
+    While `execution_time=after_pipeline` allows to consider all models for a certain eval request interval,
+    and thereby allows to select the most recent one, `execution_time=after_training` doesn't allow this.
+    This is because the models are generated during the pipeline run and the recently trained model might not be the
+    most recent model for the evaluation request that starts after the training interval. (as there might be another
+    training/model that lies between the current training and the next evaluation interval).
+    So for `execution_time=after_training` there might be multiple models that are considered `most_recent` for an
+    interval."""
```

Comment:
Just to confirm my understanding: For each model, we loop over all intervals. They are always the same for all models. However, if we have a new model, it might become the most recent model for an interval where we already evaluated a model on, and hence we evaluate a model on this interval again. So in the worst case, this setting degrades to "matrix" evaluation as well?

Reply:
For the matrix it doesn't make a big difference (just the logged `is_most_recent_model` flag might be a bit off), for non-matrix the `is_most_recent_model` is used for filtering so there might be more evaluations being launched than in the `after_pipeline` case.

Reply:
ok then I think I understood it correctly. Thanks!